tools/teal/templates/docs/delegate-keyreg.teal.md

```diff
@@ -0,0 +1,97 @@
+# Delegate Key Registration
+
+## Functionality
+
+For accounts in cold storage that still want to participate in the Algorand protocol, it is undesirable to thaw the spending key whenever new participation keys need to be registered. This contract allows you to delegate a key whose sole permission is to sign `KeyReg` transactions for its associated account.
+
+This contract is intended to be used as a "delegated" script, *not* as a contract-only account. A spending key should sign this contract.
+
+The contract is designed approve transactions that meet the following criteria:
+  1. The transaction is a `KeyReg` transaction
+  2. The transaction's `FirstValid` round is a multiple of `TMPL_PERIOD`
+  3. The transaction's `LastValid` round is within `TMPL_DUR` of its `FirstValid`
+  4. The transaction's `LastValid` is before the expiration round `TMPL_EXPIRE`
+  5. The transaction's `Lease` field is exactly `TMPL_LEASE`
+  6. The transaction's `Fee` is less than `TMPL_FEE`
+  7. The transaction's `TxID` has been signed by `TMPL_AUTH`, and a valid signature can be found in `arg_0`
+
+## Parameters:
+
+  - `TMPL_AUTH`: public key of key that can authorize `KeyReg` transactions
+  - `TMPL_EXPIRE`: round at which key expires
+  - `TMPL_PERIOD`: round multiple where a key registration period begins
+  - `TMPL_DUR`: duration of an allowed registration period
+  - `TMPL_LEASE`: transaction lease used for replay protection
+  - `TMPL_FEE`: maximum fee used by authorized transactions
+
+## Code overview
+
+### Initial checks
+First, check that this is a key registration transaction. The possible valid values of this enum may be found [here](https://github.com/algorand/go-algorand/blob/9978b3aed0643751246af82f5538ba1e7de47310/data/transactions/logic/assembler.go#L569).
+
+```
+txn TypeEnum
+int 2
+==
+```
+
+Next, check that the fee of this transaction is less than `TMPL_FEE`. Fold this check into above with a logical `AND`.
+
+```
+txn Fee
+int TMPL_FEE
+<
+&&
+```
+
+Third, check that this contract hasn't expired by checking if `LastValid` is less than `TMPL_EXPIRE`.
+
+```
+txn LastValid
+int TMPL_EXPIRE
+<
+&&
+```
+
+Next, check that this transaction is valid for exactly `TMPL_DUR` rounds by making sure that `LastValid` is equal to `FirstValid + TMPL_DUR`.
+
+```
+txn LastValid
+int TMPL_DUR
+txn FirstValid
++
+==
+&&
+```
+
+Next, check that `FirstValid` is a multiple of `TMPL_PERIOD` rounds.
+
+```
+txn FirstValid
+int TMPL_PERIOD
+%
+int 0
+==
+&&
+```
+
+Next, check that the lease field is exactly `TMPL_LEASE`.
```

Comment:
It may be useful to explain why this and the previous few checks are important for together limiting the spending power of the delegate key.

Reply:
added a comment -- what do you think?